integration/compiler-mjs-output-test.ts

```diff
@@ -8,8 +8,18 @@ let projectDir: string;
 
 test.beforeAll(async () => {
   projectDir = await createFixtureProject({
-    future: { v2_routeConvention: true },
+    config: {
+      serverModuleFormat: "esm",
+      serverBuildPath: "build/index.mjs",
+      future: { v2_routeConvention: true },
+    },
     files: {
+      // Ensure the config is valid ESM
+      "remix.config.js": js`
+        export default {
+          ...global.INJECTED_FIXTURE_REMIX_CONFIG
+        };
+      `,
```

Comment:
This feels potentially confusing to me having the config come from 2 spots.  Would it be easier to grok if we made these two approaches mutually exclusive?

Reply:
Yep, this was the part I was least sure about. I'll revisit this.